lib/improver/nowcasting/nowcasting.py

```diff
@@ -0,0 +1,196 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module with utilities required for nowcasting."""
+
+import numpy as np
+
+from cf_units import Unit
+import iris
+
+from improver.cube_combiner import CubeCombiner
+from improver.utilities.cube_checker import check_cube_coordinates
+from improver.utilities.cube_manipulation import merge_cubes
+from improver.utilities.temporal import (
+    extract_nearest_time_point, iris_time_to_datetime)
+
+
+class ApplyOrographicEnhancement(object):
+
+    """Apply orographic enhancement to precipitation rate input, either to
+     add or subtract an orographic enhancement component."""
+
+    def __init__(self, operation):
+        """Initialise class.
+
+        Args:
+            operation (str):
+                Operation (+, add, -, subtract) to apply to the incoming cubes.
+
+        Raises:
+            ValueError: Operation not supported.
+
+        """
+        possible_operations = ['+', 'add', '-', 'subtract']
+
+        if operation in possible_operations:
+            self.operation = operation
+        else:
+            msg = ("Operation '{}' not supported for combining "
+                   "precipitation rate and "
+                   "orographic enhancement.".format(operation))
+            raise ValueError(msg)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ApplyOrographicEnhancement: operation: {}>')
+        return result.format(self.operation)
+
+    @staticmethod
+    def _extract_orographic_enhancement_cube(precip_cube, oe_cubes):
+        """Extract the orographic enhancement cube with the required time
+        coordinate.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing the input precipitation fields.
+            oe_cubes (iris.cube.Cube or iris.cube.CubeList):
+                Cube or CubeList containing the orographic enhancement fields.
+
+        Returns:
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement fields at the
+                required time.
+
+        """
+        time_point, = iris_time_to_datetime(precip_cube.coord("time"))
+        oe_cube = extract_nearest_time_point(oe_cubes, time_point)
+        return oe_cube
+
+    def _apply_cube_combiner(self, precip_cube, oe_cube):
```

Comment:
Maybe `_apply_orographic_enhancement` - goal not task?

Reply:
Edited as suggested.